extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java

```diff
@@ -0,0 +1,837 @@
+            }
+        }
+        return null;
+    }
+
+    private AnnotationTarget findProperty(String property, ClassInfo clazz, IndexView index) {
+        int start = property.indexOf("(");
+        if (start != -1) {
+            // TODO validate virtual method parameters?
+            property = property.substring(0, start);
+        }
+        while (clazz != null) {
+            // Fields
+            for (FieldInfo field : clazz.fields()) {
+                if (Modifier.isPublic(field.flags()) && !Modifier.isStatic(field.flags())
+                        && !ValueResolverGenerator.isSynthetic(field.flags()) && field.name().equals(property)) {
+                    return field;
+                }
+            }
+            // Methods
+            for (MethodInfo method : clazz.methods()) {
+                if (Modifier.isPublic(method.flags()) && !Modifier.isStatic(method.flags())
+                        && !ValueResolverGenerator.isSynthetic(method.flags()) && (method.name().equals(property)
+                                || ValueResolverGenerator.getPropertyName(method.name()).equals(property))) {
+                    return method;
+                }
+            }
+            DotName superName = clazz.superName();
+            if (superName == null || DotNames.OBJECT.equals(superName)) {
+                clazz = null;
+            } else {
+                clazz = index.getClassByName(clazz.superName());
+            }
+        }
+        return null;
+    }
+
+    private void processsTemplateData(IndexView index, AnnotationInstance templateData, AnnotationTarget annotationTarget,
+            Set<ClassInfo> controlled, Map<ClassInfo, AnnotationInstance> uncontrolled) {
+        AnnotationValue targetValue = templateData.value("target");
+        if (targetValue == null || targetValue.asClass().name().equals(ValueResolverGenerator.TEMPLATE_DATA)) {
+            controlled.add(annotationTarget.asClass());
+        } else {
+            ClassInfo uncontrolledClass = index.getClassByName(targetValue.asClass().name());
+            if (uncontrolledClass != null) {
+                uncontrolled.compute(uncontrolledClass, (c, v) -> {
+                    if (v == null) {
+                        return templateData;
+                    }
+                    // Merge annotation values
+                    AnnotationValue ignoreValue = templateData.value("ignore");
+                    if (ignoreValue == null || !ignoreValue.equals(v.value("ignore"))) {
+                        ignoreValue = AnnotationValue.createArrayValue("ignore", new AnnotationValue[] {});
+                    }
+                    AnnotationValue propertiesValue = templateData.value("properties");
+                    if (propertiesValue == null || propertiesValue.equals(v.value("properties"))) {
+                        propertiesValue = AnnotationValue.createBooleanValue("properties", false);
+                    }
```

Comment:
TBH I don't remember the details and don't have the capacity to process it right now ;-). I can remove the comment but I don't think it would help anything...